token-client/src/main/java/com/sap/cloud/security/xsuaa/client/XsuaaDefaultEndpoints.java

```diff
@@ -16,25 +22,51 @@
 	private static final String AUTHORIZE_ENDPOINT = "/oauth/authorize";
 	private static final String KEYSET_ENDPOINT = "/token_keys";
 
+	private static final Logger LOGGER = LoggerFactory.getLogger(XsuaaDefaultEndpoints.class);
+
 	/**
 	 * Creates a new XsuaaDefaultEndpoints.
 	 *
 	 * @param baseUri
 	 *            - the base URI of XSUAA. Based on the base URI the tokenEndpoint,
 	 *            authorize and key set URI (JWKS) will be derived.
+	 * @deprecated gets removed with the major release 3.0.0 Use instead
+	 *             {@link #XsuaaDefaultEndpoints(OAuth2ServiceConfiguration)}
 	 */
+	@Deprecated
 	public XsuaaDefaultEndpoints(URI baseUri) {
 		assertNotNull(baseUri, "XSUAA base URI must not be null.");
+		LOGGER.debug("Xsuaa default service endpoint = {}", baseUri);
 		this.baseUri = baseUri;
 	}
 
+	/**
+	 * Creates a new XsuaaDefaultEndpoints.
+	 *
+	 * @param config
+	 *            - OAuth2ServiceConfiguration of XSUAA. Based on the
+	 *            credential-type from the configuration, the tokenEndpoint URI,
+	 *            authorize and key set URI (JWKS) will be derived.
+	 */
+	public XsuaaDefaultEndpoints(@Nonnull OAuth2ServiceConfiguration config) {
+		assertNotNull(config, "OAuth2ServiceConfiguration must not be null.");
+		if (config.getCredentialType() != null && config.getCredentialType() == CredentialType.X509) {
+			this.baseUri = config.getCertUrl();
```

Comment:
I think thats not correct. We want to call the KEYSET_ENDPOINT and the .wellknown configuration endpoint with the base url, but the AUTHORIZE_ENDPOINT with the cert url.